src/components/TodoItem.tsx

```diff
@@ -0,0 +1,92 @@
+import classNames from 'classnames';
+import { useState } from 'react';
+import { Todo } from '../types/Todo';
+import { Errors } from '../types/Errors';
+import { postDelete } from '../api/todos';
+
+type Props = {
+  todo: Todo | null;
+  setTypeError: (typeError: Errors) => void;
+  setNotificationError: (notificationError: boolean) => void;
+  todoList: Todo[];
+  setTodoList: (todoList: Todo[] | null) => void;
+  loadersTodosId: number[] | null;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  setTypeError,
+  setNotificationError,
+  todoList,
+  setTodoList,
+  loadersTodosId,
+}) => {
+  const [loaderTodo, setLoaderTodo] = useState(false);
+  const { id, title, completed } = todo || { id: 0 };
+
+  const deleteClickHandler = () => {
```

Comment:
instead of passing all props which are needed to create this function to `TodoItem` (but directly related to a separate item) component (e.g. `setTypeError`, `setTodoList`) create this function in `App.tsx` and pass it as prop

The same with `Footer` component